src/core/Akka.Remote/Endpoint.cs

```diff
@@ -320,44 +339,47 @@ internal class ReliableDeliverySupervisor : UntypedActor
         /// If we already have an inbound handle then UID is initially confirmed.
         /// (This actor is never restarted.)
         /// </summary>
-        public bool UidConfirmed { get; set; }
+        public bool UidConfirmed { get; private set; }
 
-        public Deadline BailoutAt
+        private Deadline _bailoutAt = null;
+
+        protected override SupervisorStrategy SupervisorStrategy()
         {
-            get { return Deadline.Now + _settings.InitialSysMsgDeliveryTimeout; }
+            return new OneForOneStrategy(ex =>
+            {
+                if (ex is IAssociationProblem)
+                    return Directive.Escalate;
+
+                _log.Warning("Association with remote system {0} has failed; address is now gated for {1} ms. Reason is: [{2}]", _remoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds, ex);
+                UidConfirmed = false; // Need confirmation of UID again
+                if (_bufferWasInUse)
+                {
+                    if ((_resendBuffer.Nacked.Any() || _resendBuffer.NonAcked.Any()) && _bailoutAt == null)
+                        _bailoutAt = Deadline.Now + _settings.InitialSysMsgDeliveryTimeout;
+                    Become(() => Gated(writerTerminated: false, earlyUngateRequested: false));
+                    _currentHandle = null;
+                    Context.Parent.Tell(new EndpointWriter.StoppedReading(Self));
+                    return Directive.Stop;
+                }
+
+                return Directive.Escalate;
+            });
         }
 
-        private ICancelable _autoResendTimer;
+
+        private ICancelable _maxSilenceTimer = null;
```

Comment:
New timer for silencing under quarantines - will explain more in the message-handling section for this